lib/puppet/transaction.rb

```diff
@@ -302,6 +310,13 @@ def failed_dependencies?(resource)
   # up-front at failure time because the graph may be mutated as we
   # walk it.
   def propagate_failure(resource)
+
+    provider_class = resource.provider.class
+    if(prefetch_failed_providers[resource.type][provider_class.name] and s = resource_status(resource))
```

Comment:
missing space after if
don't need the ()'s
Did you intended to assign the value of "s" in this expression? And then test if it wasn't nil
Personally I'd rather see you assign it and then s.nil?
I believe the preferred thing for doing an "and" is to use "&&"

Reply:
The assignment to `s` is intentional, but I agree that it should be done before the `if`, and in the `if` should be `!s.nil?`. Assignment in an `if` always raises the question of "was this intentional", so I try to avoid it whenever possible.

Reply:
That's fine.  I copied the if s = ... syntax from another part of the code, but going forward I will try to remember that it is considered a code smell.